src/main/java/org/folio/rest/impl/LoansAPI.java

```diff
@@ -180,6 +186,54 @@ public void postLoanStorageLoansAnonymizeByUserId(
     });
   }
 
+  @Override
+  public void postLoanStorageAnonymizeLoans(AnonymizeLoansRequest request,
+                                            Map<String, String> okapiHeaders,
+                                            Handler<AsyncResult<Response>> responseHandler,
+                                            Context vertxContext) {
+
+    AnonymizeLoansResponse response = new AnonymizeLoansResponse();
+
+    List<String> loanIds = request.getLoanIds();
+    final Set<String> invalidUuids = loanIds.stream()
+      .filter(id -> !UUIDValidation.isValidUUID(id))
+      .collect(Collectors.toSet());
+
+    if (!invalidUuids.isEmpty()) {
```

Comment:
Is it intended to cancel loans anonymization for all loans in case some of loan ids are incorrect?

Reply:
Yes

Reply:
This has been reworked.